src/annotator/anchoring/test/types-test.js

```diff
@@ -19,43 +21,290 @@ describe('types', function () {
     document.body.appendChild(container);
   });
 
-  after(function () {
+  afterEach(() => {
     container.remove();
   });
 
-  describe('TextQuoteAnchor', function () {
-    describe('#toRange', function () {
-      it('returns a valid DOM Range', function () {
+  describe('RangeAnchor', () => {
+    let fakeSniff;
+    let fakeSerializedRange;
+    let fakeNormalize;
+    let fakeSerialize;
+
+    beforeEach(() => {
+      fakeSerializedRange = sinon.stub();
+      fakeSerialize = sinon.stub();
+      fakeNormalize = sinon.stub().returns({
+        serialize: fakeSerialize,
+      });
+      fakeSniff = sinon.stub().returns({
+        normalize: fakeNormalize,
+      });
+      $imports.$mock({
+        './range': {
+          sniff: fakeSniff,
+          SerializedRange: fakeSerializedRange,
+        },
+      });
+    });
+
+    afterEach(() => {
+      $imports.$restore();
+    });
+    function createRangeAnchor() {
+      return new Range();
+    }
+
+    describe('#constructor', () => {
+      it('throws an error if `root` is missing', () => {
+        assert.throws(() => {
+          new RangeAnchor();
+        }, 'missing required parameter "root"');
+      });
+
+      it('throws an error if `range` is missing', () => {
+        assert.throws(() => {
+          new RangeAnchor({});
+        }, 'missing required parameter "range"');
+      });
+    });
+
+    describe('#fromRange', () => {
+      it('returns a new RangeAnchor instance', () => {
+        const anchor = RangeAnchor.fromRange(container, createRangeAnchor());
+        assert.calledOnce(fakeNormalize);
+        assert.instanceOf(anchor, RangeAnchor);
+      });
+    });
+
+    describe('#fromSelector', () => {
+      it('returns a new RangeAnchor instance', () => {
+        const anchor = RangeAnchor.fromSelector(container, {
+          start: container.firstChild,
+          startOffset: 0,
+          end: container.lastChild,
+          endOffset: 1,
+        });
+        assert.calledOnce(fakeSerializedRange);
+        assert.instanceOf(anchor, RangeAnchor);
+      });
+    });
+
+    describe('#toRange', () => {
+      it('returns a normalized range instance ', () => {
+        fakeNormalize.returns({
+          toRange: sinon.stub().returns('normalized range'),
+        });
+        const range = new RangeAnchor(container, createRangeAnchor());
+        assert.equal(range.toRange(), 'normalized range');
+      });
+    });
+
+    describe('#toSelector', () => {
+      beforeEach(() => {
+        fakeSerialize.returns({
+          start: container.firstChild,
```

Comment:
Using `container.firstChild` and `container.lastChild` as the values for the start and end points is confusing because in the actual app, these would be XPath-like strings. It might not actually matter for the purpose of these tests if they just copy/pass the values around, but since tests serve to help the reader understand the intent of the code as well as check correctness, I prefer that tests use realistic-looking values.

Reply:
This was an oversight on my part